femtools/Read_GMSH.F90

```diff
@@ -290,14 +339,15 @@ end function read_gmsh_simple
   ! Read through the head to decide whether binary or ASCII, and decide
   ! whether this looks like a GMSH mesh file or not.
 
-  subroutine read_header( fd, lfilename, gmshFormat )
+  subroutine read_header( fd, lfilename, gmshFormat, versionNumber )
     integer fd, gmshFormat
 
     character(len=*) :: lfilename
     character(len=longStringLen) :: charBuf
     character :: newlineChar
-    integer gmshFileType, gmshDataSize, one
-    real versionNumber
+    integer :: gmshFileType, gmshDataSize, one, i
+    logical :: decimalVersion = .false.
```

Comment:
This is dangerous (a bit of a pitfall in fortran): when using an initialisation statement in the declaration the variable automatically acquires the save attribute. This means that decimalversion is only set to .false. *once* and retains whatever it was set to in the previous call. Reading the code below, I _think_ what will happen is that if we read a "decimalversion" file first, the next file that is read it will a decimalversion regardless of whether there is a dot.

Reply:
Wow, I wasn't aware of that!

Reply:
On this topic, I found this [webpage](http://www.cs.rpi.edu/~szymansk/OOF90/bugs.html) which includes what @stephankramer explained and lists another few similar issues. I thought it was relevant to share. And indeed thanks to @stephankramer for the knowledge. :)